Firstly, the initial attempt wasn't good enough:
```
runtime/caml/platform.h: Assembler messages:
runtime/caml/platform.h:43: Error: unrecognized opcode `pause'
```
The Linux kernel uses this: https://github.com/torvalds/linux/blob/f0c4d9fc9cc9462659728d168387191387e903cc/arch/riscv/include/asm/vdso/processor.h#L24-L29

So I replaced the definition with
```c
/* Encoding of the pause instruction */
__asm__ volatile (".4byte 0x100000F");
```
and this seems to work (testsuite passes).

Secondly: there were some changes in `trunk` that had not been ported over to #11418, so in fact the RISC-V native backend is disabled in `trunk` at the moment. I made the needed tweaks in this PR (testsuite passes).

@kayceesrk do you mind doing a formal review of this PR? Thanks!

--- a/runtime/caml/platform.h
+++ b/runtime/caml/platform.h
@@ -40,6 +40,7 @@ Caml_inline void cpu_relax() {
 #elif defined(__aarch64__)
   __asm__ volatile ("yield" ::: "memory");
 #elif defined(__riscv)
+  /* Encoding of the pause instruction */
   __asm__ volatile (".4byte 0x100000F");
 #else
   /* Just a compiler barrier */

